legal-api/tests/unit/services/filings/validations/test_conversion.py

```diff
@@ -99,46 +115,83 @@ def test_invalid_nr_conversion(session):
     """Assert that party is invalid."""
+    registration_date = datetime(year=2020, month=6, day=10, hour=5, minute=55, second=13)
+    business = factory_business('FM1234567', founding_date=registration_date, last_ar_date=None,
+                                entity_type=legal_type,
+                                state=Business.State.ACTIVE)
     filing['filing']['conversion']['parties'][0]['roles'] = []
     with patch.object(NameXService, 'query_nr_number', return_value=MockResponse(nr_response)):
-            err = validate(filing)
+            err = validate(business, filing)
 
     assert err
     assert err.msg[0]['error'] == expected_msg
 
 
 @pytest.mark.parametrize(
-    'test_name, filing',
+    'test_name, legal_type, filing',
     [
-        ('sp_invalid_business_address', copy.deepcopy(SP_CONVERSION)),
-        ('gp_invalid_business_address', copy.deepcopy(GP_CONVERSION)),
+        ('sp_invalid_business_address', 'SP', copy.deepcopy(SP_CONVERSION)),
+        ('gp_invalid_business_address', 'GP', copy.deepcopy(GP_CONVERSION)),
     ]
 )
-def test_invalid_business_address(session, test_name, filing):
+def test_invalid_business_address(session, test_name, legal_type, filing):
     """Assert that delivery business address is invalid."""
+    registration_date = datetime(year=2020, month=6, day=10, hour=5, minute=55, second=13)
+    business = factory_business('FM1234567', founding_date=registration_date, last_ar_date=None,
+                                entity_type=legal_type,
+                                state=Business.State.ACTIVE)
     filing['filing']['conversion']['offices']['businessOffice']['deliveryAddress']['addressRegion'] = \
         'invalid'
     filing['filing']['conversion']['offices']['businessOffice']['deliveryAddress']['addressCountry'] = \
         'invalid'
     with patch.object(NameXService, 'query_nr_number', return_value=MockResponse(nr_response)):
-            err = validate(filing)
+            err = validate(business, filing)
 
     assert err
     assert err.msg[0]['error'] == "Address Region must be 'BC'."
     assert err.msg[1]['error'] == "Address Country must be 'CA'."
+
+
+@pytest.mark.parametrize(
+    'test_name, legal_type, start_date, filing, expected_msg',
+    [
+        ('sp_invalid_start_date', 'SP', '2016-01-01', copy.deepcopy(SP_CONVERSION),
+         'Start Date must be less than or equal to 2 years in the past and less than or equal to 90 days in the '+
+         'future from the registration date.'),
+        ('gp_invalid_start_date', 'GP', '2016-01-01', copy.deepcopy(GP_CONVERSION),
+         'Start Date must be less than or equal to 2 years in the past and less than or equal to 90 days in the future from the registration date.'),
+        ('sp_valid_start_date', 'SP', '2019-01-01', copy.deepcopy(SP_CONVERSION), None),
+        ('gp_valid_start_date', 'GP', '2019-01-01', copy.deepcopy(GP_CONVERSION), None),
+    ]
```

Comment:
can we add two test cases for the scenario where SP/GPs have a start date is more than 90 days after the registration date?

Reply:
Done